### PR TITLE
Default https

### PR DESCRIPTION
**Pull request by xurble**
Gravatar has changed so that it now recommends the https:// urls as the default

The PR reverses the default of use_ssl to be False

It also makes the domain always be www.gravatar.com since that seems to be the modern recommendation, they don't mention secure.gravatar.com any more.

https://en.gravatar.com/site/implement/images/

Finally at adds a use_ssl (default False) parameter to the get_profile method.

### CODE REVIEW

**Comment by pabluk** [comment]
Hi @xurble, thanks for your contribution! it makes sense at this time where everything is being HTTPS by default. I'll take a look tomorrow to review this changes and prepare a new release because changing the default value of `use_ssl=True` could be a breaking change for some users.

**Comment by xurble** [comment]
You should take the plunge and release a 1.0 😀

In all seriousness, though it's been a really useful library for me. Thank you.

File: README.rst [hunk 1]
```diff
@@ -32,14 +32,14 @@ Getting the user profile image::
     from libgravatar import Gravatar
     g = Gravatar('myemailaddress@example.com')
     g.get_image()
-    'http://www.gravatar.com/avatar/0bc83cb571cd1c50ba6f3e8a78ef1346'
+    'https://secure.gravatar.com/avatar/0bc83cb571cd1c50ba6f3e8a78ef1346'
```
Comment (pabluk) [inline README.rst]: I think here we only want to change the protocol `http`, but the subdomain `www` is still ok or we do need to use `secure` here too?

File: README.rst [hunk 2]
```diff
@@ -32,14 +32,14 @@ Getting the user profile image::
...
 
 Getting the profile URL::
 
     from libgravatar import Gravatar
     g = Gravatar('myemailaddress@example.com')
     g.get_profile()
-    'http://www.gravatar.com/0bc83cb571cd1c50ba6f3e8a78ef1346'
+    'https://secure.gravatar.com/0bc83cb571cd1c50ba6f3e8a78ef1346'
```
Comment (pabluk) [inline README.rst]: same here

File: libgravatar/__init__.py [hunk 3]
```diff
@@ -125,24 +125,21 @@ def get_image(self, size=DEFAULT_IMAGE_SIZE, default="", force_default=False, ra
...
             'protocol': protocol,
-            'domain': domain,
             'hash': self.email_hash,
             'extension': extension,
             'params': params,
         }
         return base_url.format(**data)
 
-    def get_profile(self, data_format=''):
+    def get_profile(self, data_format='', use_ssl=True):
```
Comment (pabluk) [inline libgravatar/__init__.py:142]: nice catch :+1:

**Review by pabluk — COMMENTED** [review]
Everything looks very good and tested! Just one question regarding the changes on README.md

**Comment by pabluk** [comment]
By the way, when reviewing this PR I spotted an old issue, doctest strings aren't being tested, so I opened and merged this [PR to fix it](https://github.com/pabluk/libgravatar/pull/10/files).

@xurble if you could rebase this branch from master and also update the protocol (`https`) into the following docstrings in order to include those tests, it would be great.

* [libgravatar/__init__.py#L75](https://github.com/xurble/libgravatar/blob/96bda3cfb7669c50c0708164f8031537bf59a720/libgravatar/__init__.py#L75)
* [libgravatar/__init__.py#L148](https://github.com/xurble/libgravatar/blob/96bda3cfb7669c50c0708164f8031537bf59a720/libgravatar/__init__.py#L148)

**Comment by pabluk** [comment]
> You should take the plunge and release a 1.0

It would be probably the case :sweat_smile:

File: README.rst (hunk 1 above)
Comment (xurble) [reply README.rst]: That's my bad.  I changed the readme before I realized that the new default was to always use www.  I should have gone back and changed 'secure' back to 'www'.

**Comment by xurble** [comment]
OK, I am terrible with pull requests.  I think that this is now up to date with your doc string test change and I've fixed the docstrings and the readme to be correct.

**Review by pabluk — APPROVED** [review]

**Comment by pabluk** [comment]
Everything looks good and ready :+1: I'll publish a new release asap.

@xurble thank you very much for your time and this great enhancement!

**Comment by pabluk** [comment]
The new release [1.0.0 ](https://github.com/pabluk/libgravatar/releases/tag/1.0.0)is now available on PyPI https://pypi.org/project/libgravatar/ :rocket:

**Pull request merged**